src/frontend/src/optimizer/plan_node/logical_agg.rs

```diff
@@ -1233,6 +1233,16 @@ impl ToBatch for LogicalAgg {
         let new_logical = self.clone_with_input(new_input);
         if self.group_key().is_empty() {
             Ok(BatchSimpleAgg::new(new_logical).into())
+        } else if self.group_key().iter().all(|group_by_idx| {
+            self.input()
```

Comment:
```suggestion
            new_input
```

Reply:
You meant new_logical right? Because new_input is already moved at this point